### PR TITLE
[CSLD-45] Block generation cases.

### PR DESCRIPTION
**Pull request by denisshevchenko**

### CODE REVIEW

File: _docs/2017-01-10-technical.md [hunk 1]
```diff
@@ -32,22 +32,34 @@ files. We have two main configuration files: for development and for production,
...
-blocks in the next epoch. Payloads from `Data` messages (along with transactions) are
-included into blocks.
+There's one node called slot-leader on each slot, and this node has right to generate
+a new block during this slot, and this block will be added to the blockchain. However
+there's no guarantee that new block will be actually generated (e.g. slot-leader can be
+offline during corresponding slot).
+
+Furthermore, slot-leader may delegate its right to another node `N`, in this case node
+`N` will have a right to generate a new block instead of slot-leader. Please note that
+such a delegation doesn't make node `N` a slot-leader.
```
Comment (volhovM) [inline _docs/2017-01-10-technical.md]: Sounds weird. It's better to say "node with delegated right is not called a slot-leader though".

File: _docs/2017-01-10-technical.md [hunk 2]
```diff
@@ -32,22 +32,34 @@ files. We have two main configuration files: for development and for production,
...
+
+It's theoretically possible to delegate slot-leader's right to multiple nodes, but it's
+**not** recommended. Moreover, we can run multiple nodes with the same key (i.e. on one computer),
+let's say nodes `A`, `B` and `C`, and if node `A` will be elected as a slot-leader, not only `A`
+will be able to generate a new block, but nodes `B` and `C` as well. It's **not** recommended to do so.
```
Comment (volhovM) [inline _docs/2017-01-10-technical.md]: Sounds scary and obscure. Better add: in this case all nodes will issue (most probably different) block and network will fork -- each other node will accept only one of these concurrent blocks. Later, fork will be eliminated.

File: _docs/2017-01-10-technical.md [hunk 3]
```diff
@@ -32,22 +32,38 @@ files. We have two main configuration files: for development and for production,
...
 
 In other words, **a slot lasts 120 seconds**, and an epoch has `10×k=600` slots
 in it, so it lasts **1200 minutes** or **20 hours**.
 
-On each slot, one *and only one* of the nodes generates a block to be added
-to the blockchain. During the epoch, nodes send each other MPC
-messages to come to the consensus as to who would be allowed to generate
-blocks in the next epoch. Payloads from `Data` messages (along with transactions) are
-included into blocks.
+There's one node called slot-leader on each slot, and this node has right to generate
```
Comment (ChShersh) [inline _docs/2017-01-10-technical.md]: Maybe add that **only** this node has the right to generate block

File: _docs/2017-01-10-technical.md [hunk 4]
```diff
@@ -32,22 +32,38 @@ files. We have two main configuration files: for development and for production,
...
+a new block during this slot, and this block will be added to the blockchain. However
+there's no guarantee that new block will be actually generated (e.g. slot-leader can be
+offline during corresponding slot).
+
+Furthermore, slot-leader may delegate its right to another node `N`, in this case node
+`N` will have a right to generate a new block instead of slot-leader. Please note that
+node `N` with delegated right isn't called a slot-leader though.
+
+It's theoretically possible to delegate slot-leader's right to multiple nodes, but it's
+**not** recommended. Moreover, we can run multiple nodes with the same key (i.e. on one computer),
```
Comment (ChShersh) [inline _docs/2017-01-10-technical.md]: Maybe clarify what is meant by sayin _not recommended_? Like just slowdown of network or undefined behavior or unpredictable behavior or something else. Well, I see note about forks but maybe add here just _by reasons explained later_.

**Pull request merged**